amclient/amclient.py

```diff
@@ -607,12 +607,50 @@ def extract_file(self):
                         file_.write(chunk)
             return response.headers
 
+    def list_location_purposes(self):
+        """List valid location purposes in the Storage Service."""
+        return {
+            "AR": "AIP_RECOVERY",
+            "AS": "AIP_STORAGE",
+            "CP": "CURRENTLY_PROCESSING",
+            "DS": "DIP_STORAGE",
+            "SD": "SWORD_DEPOSIT",
+            "SS": "STORAGE_SERVICE_INTERNAL",
+            "BL": "BACKLOG",
+            "TS": "TRANSFER_SOURCE",
+            "RP": "REPLICATOR",
+        }
+
+    def create_location(self):
+        """Create a new location in the Storage Service."""
+        if not self.location_purpose.upper() in self.list_location_purposes():
+            return {
+                "error": "location purpose not permitted",
+                "valid_purposes": self.list_location_purposes(),
+            }
+        url = "{0}/api/v2/location/".format(self.ss_url)
+        params = {
+            "description": self.location_description
```

Comment:
`black` made this a bit hard to read :grimacing:

Reply:
Yeah, I was torn as ever between fixing black's approach and leaving it. I will create a variable above to hold the value and then that should reduce the complexity introduced here.